### DISCUSSION TITLE
Problem w/ the SolarEdge API? Any help recommended!

### DISCUSSION THREAD

**Discussion by tr14d** [wene37/WeConnect-SolarManager#2] [Q&A] [answered]
2022-09-01 22:05:51,569 :: SolarManager.Service :: ERROR :: An error occured while running the service: 'connections'
Traceback (most recent call last):
  File "/home/pi/SolarManager/./main.py", line 38, in <module>
    solarManager.run()
  File "/home/pi/SolarManager/SolarManager/SolarManager.py", line 90, in run
    currentSolarState = self.solarEdge.get_current_state()
  File "/home/pi/SolarManager/SolarEdge/SolarEdge.py", line 34, in get_current_state
    for connection in apiData["connections"]:
KeyError: 'connections'

**Comment by tr14d** [comment]
Template Config File has the following entry:

[SolarEdge]
ApiUrl=https://monitoringapi.solaredge.com/site/2410315/currentPowerFlow.json?api_key=API-KEY

I assume that from the SolarEdge Monitoring Admin page the location ID of the owned solar system needs to replace the "2410315" of the template.
I also assume that "API-KEY" needs to be replaced by the generated key from the SolarEdge Monitoring Admin page.
Nevertheless, it is not working :-(

**Reply by wene37** [reply]
Sorry for the late response. I didn't see your request before.
Probably the problem is, that your key holds special characters that need to be encoded in the API-Url. Please URL-encode your key at https://www.urlencoder.org/ (or any other) before pasting it into the config file.

I will change this in next release to have three settings:
- ApiUrl with placeholders only
- ApiKey that you can copy&paste from SolarEdge Monitoring Admin page
- LocationId (currently missing; Thanks for the hint!)
Then I will do correct URL-encode in the code and you don't need to care about this.

**Reply by tr14d** [reply]
Thanks a lot! I will give it a try soon.

**Reply by wene37** [reply, marked as answer]
Just released version 0.1.2 that fixes the issue with missing location id and URL encoding
